### ISSUE TITLE
Add dev docs to document the full release and deployment process

### ISSUE THREAD

**Issue by IanButterworth** [JuliaLang/juliaup#440]
As far as I know updating juliaup on the windows store requires manual steps. It would be good to document this so that more people can update it.

**Comment by davidanthoff** [comment]
https://github.com/JuliaLang/juliaup/wiki/Release-process

**Comment by fingolfin** [comment]
I was also looking for this and did not find it. Is there any reason this is in the Wiki, instead of just being a section of the README? At the very least, the README could link to the wiki page (explicitly saying: "The release process for juliaup is described in the Wiki(LINK)".

Or perhaps a separate `MAINTAIN.md` or `README.dev.md` or so? I'd have found any of those.


BTW, even just explaining how to build it might help get more people to contribute: I made the PR to update the version to 1.8.3, but wanted to test it. In the end I figured out that I need `cargo build`, but for the absolute n00bs this already would be nice to have.

I can make a PR in this direction but I'd rather know first if it has any chance of being acceptd

**Comment by davidanthoff** [comment]
Well, I only added it late last night ;)

I don't really care where it is, the reason I put it into the wiki is that the README at the moment very clearly targets end-users, not contributors. A link in the README to the wiki sounds like a very good idea.

And yes, totally agree that the more info we have written down the better, including things like how to build etc. Most of that is standard rust/cargo stuff, but I think having a short doc that says "this is how you check for compile errors" (`cargo check`), "this is how you build" (`cargo build`), "this is how you run tests" (`cargo test`) with links to the relevant rust docs would be great.

We could also try to add some sections on the design of the whole thing.

Short version: yes, PRs/wiki edits in any of these directions would be most welcome.
